bundles/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/handler/GroheOndusSenseHandler.java

```diff
@@ -121,38 +135,50 @@ private int getBatteryStatus(Appliance appliance) {
             if (!applianceStatusOptional.isPresent()) {
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
                         "Could not load data from API.");
-                return -1;
+                return null;
             }
 
             return applianceStatusOptional.get().getBatteryStatus();
         } catch (IOException e) {
             logger.debug("Could not load appliance status", e);
         }
-        return -1;
+        return null;
     }
 
     private @Nullable ApplianceData getApplianceData(Appliance appliance) {
+        // Dates are stripped of time part inside library
         Instant yesterday = Instant.now().minus(1, ChronoUnit.DAYS);
-        Instant today = Instant.now();
+        Instant tomorrow = Instant.now().plus(1, ChronoUnit.DAYS);
```

Comment:
Consider storing `Instant.now()` in a local variable to be used for both calculations. This will ensure that both calculations are based on the same value, even if first calculation is performed before midnight and next one after.